src/lib/git/repo.ts

```diff
@@ -51,38 +52,53 @@ export async function postNewRepo({ title, description, file, owner }: any) {
 
   const contract = getWarpContract(CONTRACT_TX_ID, userSigner)
 
-  const uuid = uuidv4()
   await contract.writeInteraction({
     function: 'initialize',
     payload: {
-      id: uuid,
+      id,
       name: title,
       description,
       dataTxId: dataTxResponse.id
     }
   })
 
-  return { txResponse: dataTxResponse, id: uuid }
+  return { txResponse: dataTxResponse }
+}
+
+export async function createNewFork(data: ForkRepositoryOptions) {
+  const userSigner = new InjectedArweaveSigner(window.arweaveWallet)
+  await userSigner.setPublicKey()
+
+  const contract = getWarpContract(CONTRACT_TX_ID, userSigner)
+
+  const uuid = uuidv4()
+  await contract.writeInteraction({
+    function: 'forkRepository',
+    payload: {
+      id: uuid,
+      name: data.name,
+      description: data.description,
+      dataTxId: data.dataTxId,
+      parent: data.parent
+    }
+  })
+
+  return uuid
 }
 
 export async function postUpdatedRepo({ fs, dir, owner, id }: PostUpdatedRepoOptions) {
```

Comment:
Seems that the same issue as before of changing the branch to master when adding files to another branch is again occuring in this PR. The code to fix the issue seems to be removed in this function.